libs/model/src/community/GenerateStakeholderGroups.command.ts

```diff
@@ -0,0 +1,138 @@
+import {
+  BalanceSourceType,
+  ContractSource,
+  InvalidState,
+  ThresholdData,
+  commonProtocol,
+  schemas,
+  type Command,
+} from '@hicommonwealth/core';
+import Web3 from 'web3';
+import { models } from '../database';
+import { GroupAttributes } from '../models';
+import { getNamespace } from '../services/commonProtocol/contractHelpers';
+
+const Errors = {
+  CommunityNotFound: 'Community not found',
+  StakesNotFound: 'Stakes not found',
+  StakeholderGroup: 'Stakeholder group not found',
+  ChainNodeNotFound: 'Chain node not found',
+  NamespaceNotFound: 'Namespace not found for this name',
+};
+
+export const GenerateStakeholderGroups: Command<
+  typeof schemas.commands.GenerateStakeholderGroups
+> = () => ({
+  ...schemas.commands.GenerateStakeholderGroups,
+  auth: [],
+  body: async ({ id }) => {
+    const community = await models.Community.findByPk(id, {
+      include: [
+        {
+          model: models.Group,
+          as: 'groups',
+          where: {
+            community_id: id!,
+            is_system_managed: true,
+          },
+          required: false,
+        },
+        {
+          model: models.CommunityStake,
+          as: 'CommunityStakes',
+          where: {
+            community_id: id!,
+          },
+          required: false,
+        },
+      ],
+    });
+
+    if (!community) {
+      throw new InvalidState(Errors.CommunityNotFound);
+    }
+
+    if (community.CommunityStakes?.length === 0) {
+      throw new InvalidState(Errors.StakesNotFound);
+    }
+
+    // check which stakes need a stakeholder group
+    const stakesWithoutGroup = community.CommunityStakes!.filter((stake) => {
```

Comment:
let's always explore the opportunity to load the partial aggregate needed to validate the command in just "one" sql call.

Reply:
consolidated